tests/integration/conftest.py

```diff
@@ -46,40 +53,78 @@ def shared_kubeflow_profile(
         The input repository with a local kubeflow stack provisioned for the
         module active profile.
     """
+    from zenml.container_registries import DefaultContainerRegistry
+    from zenml.integrations.kubeflow.metadata_stores import (
+        KubeflowMetadataStore,
+    )
     from zenml.integrations.kubeflow.orchestrators import KubeflowOrchestrator
-
-    # Patch the UI daemon as forking doesn't work well with pytest
+    from zenml.integrations.s3.artifact_stores import S3ArtifactStore
+
+    def run_in_thread(
+        daemon_function: F,
+        *args: Any,
+        pid_file: str,
+        log_file: Optional[str] = None,
+        working_directory: str = "/",
+        **kwargs: Any
+    ):
+        # run the actual passed function in a daemon thread
+        t = threading.Thread(target=daemon_function, args=args, kwargs=kwargs)
+        t.setDaemon(True)
+        t.start()
+
+        # As we are creating a daemon thread instead of a real daemon process,
+        #  no pid file is created, so we now need to pretend the pid file exists
+        def fake_pid_file_exists(pid_file: str):
+            return True
+
+        module_mocker.patch(
```

Comment:
Yes, this ensures, that the method is patched only as from the second time that it is called. For the first call it should return false so that the daemon thread can run and pretend to have written a pid file

Reply:
Ah that makes sense, thanks! I have one follow up-question, to explain let me quickly describe how I think the tests are running currently (and correct me if I'm wrong here, as then the question probably doesn't make sense 😄):
- All the examples get "started" from the same python process, which uses the module-level fixture `shared_kubeflow_profile`.
- The example copies and updates the stack with additional components that it needs.
- The example gets run and the copied stack etc get deleted.

Now to my questions: Do we call `zenml stack up` or the code equivalent at some point after copying and updating the stack to make sure the newly added components also get a chance to work?

If we don't call it: Should we?
If we do call it: Due to using a `module_mocker` to patch the logic of writing the PID file, now the thing you described above does not work anymore as it's already patched right?